bundle-workflow/src/test_workflow/dependency_installer.py

```diff
@@ -0,0 +1,65 @@
+import os
+import shutil
+
+from aws.s3_bucket import S3Bucket
+
+
+class DependencyInstaller:
+    """
+    Provides a dependency installer for the test suites.
+    """
+
+    def __init__(self, build):
+        self.build_id = build.id
+        self.version = build.version
+        self.arch = build.architecture
+        self.s3_maven_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/maven/org/opensearch"
+        )
+        self.s3_build_location = (
+            f"/builds/{self.version}/{self.build_id}/{self.arch}/plugins"
+        )
+
+    def install_maven_dependencies(self, dependency_dict):
+        """
+        Downloads the maven dependencies from S3 and puts them on the maven local path
+        for each dependency in the dependency_list.
+
+        :param dependency_dict: list of dependency names with version for which the maven artifacts need to be downloaded.
+        Example: {'opensearch-job-scheduler':'1.1.0.0', 'opensearch-core':'1.1.0'}
+        """
+        s3_bucket = S3Bucket(self.s3_maven_location)
+        for dependency, version in dependency_dict.items():
+            s3_path = f"{dependency}/{version}"
+            maven_local_path = os.path.join(
+                os.path.expanduser("~"),
+                f".m2/repository/org/opensearch/{dependency}/{version}/",
+            )
+            s3_bucket.download_folder(s3_path, maven_local_path)
+
+    def install_build_dependencies(self, dependency_dict, custom_local_path):
+        """
+        Downloads the build dependencies from S3 and puts them on the given custom path
+        for each dependency in the dependency_list.
+
+        :param dependency_list: list of dependency names with version for which the build artifacts need to be downloaded.
+        Example: {'opensearch-job-scheduler':'1.1.0.0'}
+        :param custom_local_path: the path where the downloaded dependencies need to copied.
+        """
+        s3_bucket = S3Bucket(self.s3_build_location)
+        for dependency, version in dependency_dict.items():
+            s3_path = f"{dependency}-{version}.zip"
+            s3_bucket.download_file(s3_path, custom_local_path)
+
+    def cleanup(self, local_path):
```

Comment:
This deletes a folder, which is ... dangerous if we get it wrong. For maven dependencies I don't think we will ever want to be deleting them after downloading explicitly because they will be intermixed with other dependencies that may have been potentially dropped into `~m2`. We already are doing cleanup in https://github.com/opensearch-project/opensearch-build/pull/403. I think you should delete it from the PR and close any issues that say otherwise.

Reply:
@dblock - valid point. But I think we still need to cleanup dependencies which are synced as part of one test run, else we will unnecessarily bloat the local maven over a period of time. IMO we should switch back to the custom maven local path, based on a random tmp location, where we download dependencies and cleanup after the test run. This keeps the concurrent test runs isolated. We planned on doing it earlier but then switched to having docker provide that isolation by mounting on a randomized tmp path. I think we need it in the code as developers will run this code on local setup as well where docker mounting won't provide that sort of isolation. @VachaShah and I discussed about it yesterday.

Reply:
>bloat the local maven over a period of time.

Jenkins Agent nodes will only exist for a single workflow then are torn down, how fast is this bloat?  Enough that we need to clean it out to complete a workflow?

Reply:
For the record CI will be providing throw away containers with their own file system for these jobs.

Reply:
Removed the cleanup logic.